source/_components/cover.mysensors.markdown

```diff
@@ -0,0 +1,152 @@
+
+#include <MySensors.h>
+#define SN "Cover"
+#define SV "1.1"
+
+// Actuators for moving the cover up and down respectively.
+#define COVER_UP_ACTUATOR_PIN 2
+#define COVER_DOWN_ACTUATOR_PIN 3
+// Sensors for finding out when the cover has reached its up/down position.
+// These could be simple buttons or linear hall sensors.
+#define COVER_UP_SENSOR_PIN 4
+#define COVER_DOWN_SENSOR_PIN 5
+
+#define CHILD_ID 0
+
+// Internal representation of the cover state.
+enum State {
+  IDLE,
+  UP, // Window covering. Up.
+  DOWN, // Window covering. Down.
+};
+
+static int state = IDLE;
+static int status = 0; // 0=cover is down, 1=cover is up
+MyMessage upMessage(CHILD_ID, V_UP);
+MyMessage downMessage(CHILD_ID, V_DOWN);
+MyMessage stopMessage(CHILD_ID, V_STOP);
+MyMessage statusMessage(CHILD_ID, V_STATUS);
+
+void sendState() {
+  // Send current state and status to gateway.
+  send(upMessage.set(state == UP));
+  send(downMessage.set(state == DOWN));
+  send(stopMessage.set(state == IDLE));
+  send(statusMessage.set(status));
+}
+
+void setup() {
+  pinMode(COVER_UP_SENSOR_PIN, INPUT);
+  pinMode(COVER_DOWN_SENSOR_PIN, INPUT);
+
+  // Send initial values.
+  sendState();
+}
+
+void presentation() {
+  sendSketchInfo(SN, SV);
+
+  present(CHILD_ID, S_COVER);
+}
+
+void loop() {
+  if (state == IDLE) {
+    digitalWrite(COVER_UP_ACTUATOR_PIN, LOW);
+    digitalWrite(COVER_DOWN_ACTUATOR_PIN, LOW);
+  }
+
+  if (state == UP && digitalRead(COVER_UP_SENSOR_PIN) == HIGH) {
+    Serial.println("Cover is up.");
+    // Update status and state; send it to the gateway.
+    status = 1;
+    state = IDLE;
+    sendState();
+    // Actuators will be disabled in next loop() iteration.
+  }
+
+  if (state == DOWN && digitalRead(COVER_DOWN_SENSOR_PIN) == HIGH) {
+    Serial.println("Cover is down.");
+    // Update status and state; send it to the gateway.
+    status = 0;
+    state = IDLE;
+    sendState();
+    // Actuators will be disabled in next loop() iteration.
+  }
+}
+
+void receive(const MyMessage &message) {
+  if (message.type == V_UP) {
+    // Set state to covering up and send it back to the gateway.
+    state = UP;
+    sendState();
+    Serial.println("Moving cover up.");
+
+    // Activate actuator until the sensor returns HIGH in loop().
+    digitalWrite(COVER_UP_ACTUATOR_PIN, HIGH);
+  }
+
+  if (message.type == V_DOWN) {
+    // Set state to covering up and send it back to the gateway.
+    state = DOWN;
+    sendState();
+    Serial.println("Moving cover down.");
+    // Activate actuator until the sensor returns HIGH in loop().
+    digitalWrite(COVER_DOWN_ACTUATOR_PIN, HIGH);
+  }
+
+  if (message.type == V_STOP) {
+    // Set state to idle and send it back to the gateway.
+    state = IDLE;
+    sendState();
+    Serial.println("Stopping cover.");
+
+    // Actuators will be switched off in loop().
+  }
+}
+```
+
+[main component]: /components/mysensors/
+[serial api]: https://www.mysensors.org/download/serial_api_15
```

Comment:
This link is not working anymore, cause they removed that site. Use API 2.0 instead:
https://www.mysensors.org/download/serial_api_20